traverson4j-hc5/src/main/java/uk/co/autotrader/traverson/http/ApacheHttpUriConverter.java

```diff
@@ -25,11 +26,11 @@ public ApacheHttpUriConverter(BodyFactory bodyFactory, TemplateUriUtils template
         this.conversionService = conversionService;
     }
 
-    public HttpUriRequest toRequest(Request request) {
+    public ClassicHttpRequest toRequest(Request request) {
         Map<String, List<String>> templateParams = request.getTemplateParams();
         String uri = templateUriUtils.expandTemplateUri(request.getUrl(), templateParams);
 
-        RequestBuilder requestBuilder = RequestBuilder.create(request.getMethod().name()).setUri(uri);
+        ClassicRequestBuilder requestBuilder = ClassicRequestBuilder.create(request.getMethod().name()).setUri(uri);
```

Comment:
Is this Classic idea going to stay in V5 or is it a deprecated thing to help transition between v4 and v5?

Reply:
Its not marked as deprecated so assume its here to stay?

Reply:
Just digging around the release notes, we have come across 
```
This is the first BETA release of HttpClient 5.0. The 5.0 release serices introduces support for the HTTP/2 protocol and event driven messaging APIs consistent for all supported HTTP protocol versions.

HttpClient ships with two client implementations:

HttpClient Classic is based on the classic (blocking) I/O model; largely compatible with the 4.x APIs; supports HTTP/1.1 only.
HttpClient Async is based on NIO model; new event driven APIs consistent for all supported HTTP protocol versions; supports both HTTP/1.1 and HTTP/2.
````
Would this mean we should favour the async modules to support http/2?